File: code/generate_strategies/IR(pseudocode)/cpp/cpp/function_sklearn_OneHotEncoder.cpp
#include <Eigen/Dense>
#include <stdexcept>
#include <string>
#include <sstream>
#include <vector>
#include <unordered_map>

class OneHotEncoderFunction {
public:
    static std::string test_one_hot_encoding(const std::string& data) {
        if (data.empty()) {
            throw std::invalid_argument("Input data must not be empty.");
        }

        std::istringstream iss(data);
        std::string categorical_data;
        int numeric_data;

        if (!(iss >> categorical_data >> numeric_data)) {
            throw std::invalid_argument("The input data has incorrect structure.");
        }

        
        std::unordered_map<std::string, int> category_map = {{"category1", 1}};
        int encoded_value = category_map[categorical_data];

        std::ostringstream oss;
        oss << numeric_data << "," << encoded_value;
        return oss.str();
    }
};

File: manual/code/generate_strategies/IR(summary)/java/java/function_sklearn_OneHotEncoder.java
package com.example;

import org.apache.commons.math3.linear.MatrixUtils;
import org.apache.commons.math3.linear.RealMatrix;

public class OneHotEncoderFunction {
    public String test_one_hot_encoding(String data) {
        
        class TypeError extends RuntimeException {
            public TypeError(String message) {
                super(message);
            }
        }

        class IndexError extends RuntimeException {
            public IndexError(String message) {
                super(message);
            }
        }

        
        if (data == null) {
            throw new TypeError("Input data must be a string.");
        }

        
        String[] parts = data.split(" ");
        if (parts.length < 2) {
            throw new IndexError("The input data has incorrect structure.");
        }

        String categoricalData;
        int numericData;
        try {
            
            categoricalData = parts[0];
            numericData = Integer.parseInt(parts[1]);
        } catch (Exception e) {
            throw new IndexError("Data structure is incorrect. Maybe missing categorical or numeric part.");
        }

        
        
        RealMatrix encodedMatrix = MatrixUtils.createRealMatrix(new double[][]{{1.0}});
        double encodedValue = encodedMatrix.getEntry(0, 0);

        
        String finalData = numericData + "," + encodedValue;
        return finalData;
    }
}